backend/api/services.py

```diff
@@ -215,3 +221,172 @@ def get_reddit_posts(subreddit, symb, time="week") -> pd.DataFrame:
     df.replace("", np.nan, inplace=True)
     df.dropna(inplace=True)
     return df
+
+
+# TWEETS
+def get_tweets(query) -> pd.DataFrame:
+    """Returns a dataframe of tweets matching a given search query.
+
+    Args:
+        query (str): Search query eg. AAPL
+
+    Returns:
+        pd.DataFrame: DataFrame of tweets and other relevant information.
+    """
+    query = f"\\${query} lang:en"
+    tweets = twitter.search_recent_tweets(
+        query=query,
+        tweet_fields=["author_id", "created_at", "public_metrics", "entities"],
+        max_results=100,
+    )
+
+    tweets_df = pd.DataFrame(tweets.data)
+
+    # explode out public metrics
+    pub_metrics = tweets_df["public_metrics"].apply(pd.Series)
```

Comment:
Not sure the apply part is strictly necessary since a column of a `DataFrame` is a `Series`. If you want it to not be a view of the original, you can use the `.copy()` method. 

![image](https://user-images.githubusercontent.com/13448687/202839026-607dd4d8-c283-4fa9-a6f5-2e60dd594247.png)